flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/NettyShuffleMaster.java

```diff
@@ -49,6 +57,10 @@ public class NettyShuffleMaster implements ShuffleMaster<NettyShuffleDescriptor>
 
     private final int networkBufferSize;
 
+    private final ResourceRegistry resourceRegistry;
```

Comment:
This is never used outside the constructor of `NettyShuffleMaster`.  Then why keeping the reference.

Reply:
And if we need to keep them, it might be better to have a wrapper class to hold all the tiered shuffle related components.

Reply:
Added a `TieredInternalShuffleMaster` wrapper class.
All the tiered storage operations with the shuffle master should be wrapped in this class.

The `ResourceRegistry ` is changed as a variable in the constructor of `TieredInternalShuffleMaster`.